src/Routing/SystemWebRoutingMiddleware.cs

```diff
@@ -0,0 +1,24 @@
+// MIT License.
+
+using System.Web.Routing;
+using Microsoft.AspNetCore.SystemWebAdapters.Features;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+internal sealed class SystemWebRoutingMiddleware(AspNetCore.Http.RequestDelegate next, RouteCollection routes)
+{
+    public Task InvokeAsync(HttpContextCore context)
+    {
+        var feature = new RouteDataFeature(context);
+        context.Features.Set<IRouteDataFeature>(feature);
+
+        if (routes.TryGetMapped(context.Request.Path, out var newPath, out var kv))
+        {
+            context.Request.Path = newPath;
+            feature.RouteData = new(new(kv));
```

Comment:
so if I understand, the objective is set with every invocation rather than on demand...

Reply:
correct - so we can keep the two worlds in sync - two features that are implemented together